agent-shell.el

```diff
@@ -234,8 +259,51 @@ AUTHENTICATE-REQUEST-MAKER."
 (defvar-local agent-shell--state
     (agent-shell--make-state))
 
+(defvar-local agent-shell--transcript-file nil
+  "Path to the transcript file for this buffer.")
+
 (defvar agent-shell--shell-maker-config nil)
 
+(defun agent-shell--transcript-directory ()
+  "Return the full path to the transcript directory for the current project."
+  (expand-file-name agent-shell-transcript-directory (agent-shell-cwd)))
+
+(defun agent-shell--init-transcript ()
+  "Initialize a new transcript file for this buffer.
+Returns the path to the transcript file."
+  (when agent-shell-auto-save-transcript
+    (condition-case err
+        (let* ((dir (agent-shell--transcript-directory))
+               (filename (funcall agent-shell-transcript-filename-function))
+               (filepath (expand-file-name filename dir)))
+          (make-directory dir t)
+          (write-region
+           (format "Agent Shell Transcript\nStarted: %s\nWorking Directory: %s\n\n%s\n\n"
+                   (format-time-string "%Y-%m-%d %H:%M:%S")
+                   (agent-shell-cwd)
+                   (make-string 60 ?=))
+           nil filepath)
+          filepath)
+      (error
+       (message "Failed to initialize transcript: %S" err)
+       nil))))
+
+(defun agent-shell--append-to-transcript (text &optional buffer)
```

Comment:
Let's swap buffer for file-path to make things more explicit.

cl-defun agent-shell--append-transcript (&key text file-path)